File: JAVA/633. Sum of Square Numbers.java
class Solution {
    public boolean judgeSquareSum(int c) {
        if(c==0) return true;
        int a=(int)Math.sqrt(c);
        int b=0;
        int s=0;
        while(a!=0){
            b=c-a*a;
            s=(int)Math.sqrt(b);
            if(s*s==b) return true;
            else a--;       
        }
        return false;        
    }
}

File: C++/633. Sum of Square Numbers.cpp
class Solution {
public:
    bool judgeSquareSum(int c) {
        long a=0;
        long b=sqrt(c);
        while(a<=b){
            long squareSum= a*a + b*b;
            if(squareSum==c){
                return true;
            }
            else if(squareSum<c){
                a++;
            }
            else{
                b--;
            }
        }
        return false;
    }
};
